src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -1017,8 +1041,8 @@ protected long logEvent(String eventType, JSONObject eventProperties, JSONObject
             }
 
             JSONObject library = new JSONObject();
-            library.put("name", Constants.LIBRARY);
-            library.put("version", Constants.VERSION);
+            library.put("name", this.libraryName == null ? Constants.LIBRARY_UNKNOWN : this.libraryName);
```

Comment:
nonblocking suggestion: in the `setLibraryName`, if the input value is `null`, then you can do `this.libraryName = LIBRARY_UNKNOWN`, and then you don't have to do this comparison every time

Reply:
My thought is to leave it as is. Keep it original. 

Both are fine.